src/panoptes/pocs/focuser/focuser.py

```diff
@@ -343,49 +340,44 @@ def _autofocus(self,
         start_time = current_time(flatten=True)
-        file_path_root = os.path.join(image_dir,
-                                      'focus',
-                                      self._camera.uid,
-                                      start_time)
+        file_path_root = os.path.join(image_dir, 'focus', self._camera.uid, start_time)
 
         self._autofocus_error = None
 
-        dark_thumb = None
+        dark_thumbnail = None
         if take_dark:
-            dark_path = os.path.join(file_path_root,
-                                     '{}.{}'.format('dark', self._camera.file_extension))
-            self.logger.debug('Taking dark frame {} on camera {}'.format(dark_path, self._camera))
+            dark_path = os.path.join(file_path_root, f'dark.{self._camera.file_extension}')
+            self.logger.debug(f'Taking dark frame {dark_path} on camera {self._camera}')
             try:
-                dark_thumb = self._camera.get_thumbnail(seconds,
-                                                        dark_path,
-                                                        thumbnail_size,
-                                                        keep_file=True,
-                                                        dark=True)
+                dark_thumbnail = self._camera.get_thumbnail(seconds,
+                                                            dark_path,
+                                                            thumbnail_size,
+                                                            keep_file=True,
+                                                            dark=True)
                 # Mask 'saturated' with a low threshold to remove hot pixels
-                dark_thumb = mask_saturated(dark_thumb,
-                                            threshold=0.3,
-                                            bit_depth=self.camera.bit_depth)
+                dark_thumbnail = mask_saturated(dark_thumbnail,
+                                                threshold=0.3,
+                                                bit_depth=self.camera.bit_depth)
             except Exception as err:
                 self.logger.error(f"Error taking dark frame: {err!r}")
                 self._autofocus_error = repr(err)
                 focus_event.set()
                 raise err
 
         # Take an image before focusing, grab a thumbnail from the centre and add it to the plot
-        initial_fn = "{}_{}_{}.{}".format(initial_focus,
-                                          focus_type,
-                                          "initial",
-                                          self._camera.file_extension)
+        initial_fn = f"{initial_focus}-{focus_type}-initial.{self._camera.file_extension}"
         initial_path = os.path.join(file_path_root, initial_fn)
 
         try:
             initial_thumbnail = self._camera.get_thumbnail(seconds, initial_path, thumbnail_size, keep_file=True)
```

Comment:
We can probably clean up this initial thumbnail code by moving it to the main focusing loop below.

Reply:
This takes a picture before any movement of the focuser, whereas the first step in the loop is to move to a new position.

You're right that it probably could be cleaned up a bit but I'm going to leave for now.